### PR TITLE
[Contributing Guidelines] Add script to check that versions are present

### PR DESCRIPTION
**Pull request by ericphanson**
This can help with PRs like https://github.com/JuliaRegistries/General/pull/75319

Based on @GunnarFarneback's script from https://github.com/JuliaRegistries/General/pull/35965#issuecomment-832721704, but updated to use RegistryInstances many folks don't have git registries. (Also should be usable for other registries by changing the `registry_uuid`; not super relevant here but I know folks often copy-paste stuff from General in order to setup other registries).

### CODE REVIEW

**Comment by giordano** [comment]
```julia
julia> check_package_versions("FastParzenWindows", "https://github.com/ngiann/FastParzenWindows.jl.git");
Cloning into '/var/folders/v2/hmy3kzgj4tb3xsy8qkltxd0r0000gn/T/jl_CBJohl'...
remote: Enumerating objects: 261, done.
remote: Counting objects: 100% (79/79), done.
remote: Compressing objects: 100% (68/68), done.
remote: Total 261 (delta 40), reused 26 (delta 10), pack-reused 182
Receiving objects: 100% (261/261), 98.03 KiB | 760.00 KiB/s, done.
Resolving deltas: 100% (142/142), done.
ERROR: UndefVarError: `pretty_print` not defined
Stacktrace:
 [1] check_packages_versions(pkg_names::Vector{String}, repo_url::String; registry_uuid::UUID, verbose::Bool, throw::Bool)
   @ Main /tmp/foo.jl:33
 [2] check_packages_versions
   @ /tmp/foo.jl:15 [inlined]
 [3] #check_package_versions#8
   @ /tmp/foo.jl:38 [inlined]
 [4] check_package_versions(pkg_name::String, repo_url::String)
   @ Main /tmp/foo.jl:38
 [5] top-level scope
   @ REPL[8]:1
```

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -89,6 +90,67 @@ You generally should not merge your own registrations (though you can make reque
...
+        pkg = registry.pkgs[only(uuids_from_name(registry, pkg_name))]
+        versions = registry_info(pkg).version_info
+        for version in sort(collect(keys(versions)))
+            tree_sha = versions[version].git_tree_sha1
+            found = success(`$(git()) -C $dir rev-parse -q --verify "$(tree_sha)^{tree}"`)
+
+            push!(table, (; pkg_name, version, found))
+        end
+    end
+    verbose && pretty_print(table)
```
Comment (giordano) [inline CONTRIBUTING.md:130]: Was this meant to be
```suggestion
    verbose && pretty_print_table(table)
```
?

Comment (ericphanson) [reply CONTRIBUTING.md:130]: Oops, yes

Comment (ericphanson) [reply CONTRIBUTING.md]: I had tested it, just not in a fresh session, and I still had an old method in my session. Tested now in a fresh session & it should be OK

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -89,6 +90,67 @@ You generally should not merge your own registrations (though you can make reque
...
+    end
+    verbose && pretty_print_table(table)
+    throw && check_all_found(table)
+    return table
+end
+
+check_package_versions(pkg_name, repo_url; kw...) = check_packages_versions([pkg_name], repo_url; kw...)
+```
+
+For example, in https://github.com/JuliaRegistries/General/pull/75319, a package author wanted to update the URL associated
```
Comment (ericphanson) [inline CONTRIBUTING.md:138]:
```suggestion
For example, in [General#75319](https://github.com/JuliaRegistries/General/pull/75319), a package author wanted to update the URL associated
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -89,6 +90,67 @@ You generally should not merge your own registrations (though you can make reque
...
+to their package "FastParzenWindows". At the time, the package had 1 registered version. We can check that it is present in the new repository via:
+```julia
+julia> check_package_versions("FastParzenWindows", "https://github.com/ngiann/FastParzenWindows.jl.git");
+Cloning into '/var/folders/jb/plyyfc_d2bz195_0rc0n_zcw0000gp/T/jl_ke9E8C'...
+...text omitted...
+FastParzenWindows: v0.1.2 found
+```
+We see that this version was found in the new repository. This script was based on https://github.com/JuliaRegistries/General/pull/35965#issuecomment-832721704,
```
Comment (ericphanson) [inline CONTRIBUTING.md:146]:
```suggestion
We see that this version was found in the new repository. This script was based on [this comment from General#35965](https://github.com/JuliaRegistries/General/pull/35965#issuecomment-832721704),
```

**Review by giordano — APPROVED** [review]
I confirm it works well now, thanks, this is very useful to add to the README!

Side question: Is there any package where this snippet could live, to make this easier to test instead of copying a bunch of functions?  But this can be done as a follow up.

**Comment by ericphanson** [comment]
I’m not sure. It would be nice to somehow automate this check, maybe as another workflow which runs on all PRs and detects URL changes. Maybe with the code in RegistryCI? And another workflow so it can get triggered on manual PRs.

**Comment by ericphanson** [comment]
Oops, clicked the wrong button. I think it would make sense to put the code in RegistryCI. It wouldn’t be part of automerge but I think it could make sense to put it in the package anyway. Then also we could later try to add another workflow to run the code on PRs with URL changes.

**Comment by giordano** [comment]
> It would be nice to somehow automate this check, maybe as another workflow which runs on all PRs and detects URL changes. Maybe with the code in RegistryCI? 

Yes, that'd be cool

**Comment by ericphanson** [comment]
Let’s put it here for now though and remove it if it gets moved somewhere better.

**Pull request merged**